### PR TITLE
feat: add phi-3-mini series

### LINKED ISSUE #1371: FEAT: support phi-3 model

### PR DESCRIPTION
**Pull request by orangeclk**
Resolve #1371

Add phi-3-mini-128k-instruct and phi-3-mini-4k-instruct.

### CODE REVIEW

File: xinference/model/llm/llm_family.json [hunk 1]
```diff
@@ -461,6 +461,66 @@
...
   },
+  {
+    "version": 1,
+    "context_length": 128000,
+    "model_name": "phi-3-mini-128k-instruct",
+    "model_lang": [
+      "en"
+    ],
+    "model_ability": [
+      "generate"
```
Comment (qinxuye) [inline xinference/model/llm/llm_family.json:472]: Actually, this model is a chat model, so you need to add chat ability, hence, some fields might be absent, like https://github.com/qinxuye/inference/blob/78897c51ca7c0a4751a870dd0b8b6813a7eafd58/xinference/model/llm/llm_family.json#L1346-L1361

Comment (orangeclk) [reply xinference/model/llm/llm_family.json:472]: Thank you. I will try to add chat ability.

Comment (qinxuye) [reply xinference/model/llm/llm_family.json:472]: You can refer to #1384 , this is another contribution from the community.

**Comment by qinxuye** [comment]
Can you paste a screenshot to show the chat?

**Comment by orangeclk** [comment]
> Can you paste a screenshot to show the chat?

image
image
image

The gguf version has a worse performance.

**Comment by qinxuye** [comment]
> > Can you paste a screenshot to show the chat?
> 
> image image image
> 
> The gguf version has a worse performance.

Really nice job, I noticed the modelscope has the model, can you update the llm_family_modelscope.json as well?

**Comment by qinxuye** [comment]
>  image
> 
Looks like the gguf model didn't stop correctly.

**Comment by orangeclk** [comment]
> > image
> 
> Looks like the gguf model didn't stop correctly.

No, it didn't. I have made a change here. Now it works.
image

File: xinference/model/llm/llm_family.json [hunk 2]
```diff
@@ -461,6 +461,102 @@
...
+      ],
+      "intra_message_sep": "\n",
+      "inter_message_sep": "<|end|>\n",
+      "stop_token_ids":[
+        32000,
+        32001,
+        32007
+      ],
+      "stop": [
+        "<|end|>\n"
```
Comment (qinxuye) [inline xinference/model/llm/llm_family.json:556]: Is it better to be `<|end|>` ? From https://huggingface.co/microsoft/Phi-3-mini-128k-instruct/blob/main/tokenizer_config.json it's 32007.

Comment (qinxuye) [reply xinference/model/llm/llm_family.json:556]: And is it right that there is one stop ,but with 3 stop ids?

**Review by qinxuye — COMMENTED** [review]
I have some further questions.

File: xinference/model/llm/llm_family.json (hunk 2 above)
Comment (orangeclk) [reply xinference/model/llm/llm_family.json:556]: > Is it better to be `<|end|>` ? From https://huggingface.co/microsoft/Phi-3-mini-128k-instruct/blob/main/tokenizer_config.json it's 32007.

I've found this problem too. My local version has been modified and I'm about to commit it.

Comment (orangeclk) [reply xinference/model/llm/llm_family.json:556]: > And is it right that there is one stop ,but with 3 stop ids?

I got the three stop token ids from here https://huggingface.co/microsoft/Phi-3-mini-4k-instruct/blob/main/generation_config.json. However, I think you are right. I should follow the link you show to me.

Comment (qinxuye) [reply xinference/model/llm/llm_family.json:556]: Yeah, you can add the corresponding tokens to stop.

**Comment by orangeclk** [comment]
> Really nice job, I noticed the modelscope has the model, can you update the llm_family_modelscope.json as well?

added

**Comment by qinxuye** [comment]
Really nice work. Could you run gen_docs.py again to update the modelscope info?

**Comment by qinxuye** [comment]
I did it for you, btw, the docs can be generated by running `doc/source/gen_docs.py`.

Anyway, thanks for you contribution, well done.

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**